system/HTTP/CURLRequest.php

```diff
@@ -651,11 +651,11 @@ protected function setCURLOptions(array $curlOptions = [], array $config = [])
 
         // version
         if (! empty($config['version'])) {
-            if ($config['version'] === 1.0) {
+            if ($config['version'] === 1.0 || $config['version'] === '1.0') {
                 $curlOptions[CURLOPT_HTTP_VERSION] = CURL_HTTP_VERSION_1_0;
-            } elseif ($config['version'] === 1.1) {
+            } elseif ($config['version'] === 1.1 || $config['version'] === '1.1') {
                 $curlOptions[CURLOPT_HTTP_VERSION] = CURL_HTTP_VERSION_1_1;
-            } elseif ($config['version'] === 2.0) {
+            } elseif ($config['version'] === 2.0 || $config['version'] === '2.0') {
                 $curlOptions[CURLOPT_HTTP_VERSION] = CURL_HTTP_VERSION_2_0;
```

Comment:
Can't we just cast it to `string` or `float` and do only one check?

Reply:
Glad for the question - that was my first thought, but checking the rest of the code in the Library having all strict comparisons made me reluctant. Would `(string)$config['version'] === '1.0'` have any side-effects? is float-to-string cast always locale-independant?

Reply:
I don't believe it will cause any problems, we use this config value to set this one option and nothing else.
Casting to string should not have any side effects in this case.

Reply:
But.... seems like `sprintf()` should be used - https://www.php.net/manual/en/function.strval.php

"This function performs no formatting on the returned value. If you are looking for a way to format a numeric value as a string, please see [sprintf()](https://www.php.net/manual/en/function.sprintf.php) or [number_format()](https://www.php.net/manual/en/function.number-format.php)."

Edge-cases.... brrrr.....

Reply:
I was quite surprised by the original code actually. Is `float 1.0` always exactly 1.0?

Reply:
Yes, you're right. I guess we should use something like: 
`$version = sprintf('%.1f', $config['version'])` at the beginning. That should be safe enough.

> I was quite surprised by the original code actually. Is float 1.0 always exactly 1.0?

Yes, it's safe in our case. The problem may occur when we start doing operations on float value.

Reply:
Pushed a new commit with that conversion

Reply:
Seems to be ok...
```php
<?php

$config['version'] = '1.0';
$version = sprintf('%.1f', $config['version']);
var_dump($version);

$config['version'] = 1.0;
$version = sprintf('%.1f', $config['version']);
var_dump($version);


/* returns:

/opt/project/cast.php:5:
string(3) "1.0"
/opt/project/cast.php:9:
string(3) "1.0"

*/
```

Reply:
As far as I know, `(string)` is locale-independent. But `var_dump((string) 0.00001);` returns  `"1.0E-5"`.